src/automata.jl

```diff
@@ -294,7 +297,7 @@ function makeindexautomaton(rws::RewritingSystem, abt::Alphabet=alphabet(orderin
             end
         end
         i += 1
-        indcs = findall(isequal(i), Σᵢ)
+        indcs = findall(isequal(i), stateslengths(a))
```

Comment:
```julia
resize!(idcs, 0)
for (k,len) in enumerate(stateslengths(a))
    len == i && push!(idcs, k)
end
```

Reply:
also above: I think that `τ = last(walk(a, @view(name(σ)[2:end])))` should cut allocations as well

Reply:
This suggestion is good, but it has to be applied in line 308 (in line 300 we first declare `indcs`).